[CI] MixedClusterClientYamlTestSuiteIT test {p0=search.vectors/40_knn_search/Dimensions are dynamically set} failing

**Build Scans:**
- [elasticsearch-periodic #9192 / openjdk21_8.19.0_java-fips-matrix-bwc](https://gradle-enterprise.elastic.co/s/hbs6ydrpi5on2)
- [elasticsearch-pull-request #79148 / 9.0.4_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/prj7ghirb3ixo)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.19.0#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/40_knn_search/Dimensions are dynamically set}" -Dtests.seed=7CBBD99F36990070 -Dtests.bwc=true -Dtests.locale=sc-Latn-IT -Dtests.timezone=Pacific/Ponape -Druntime.java=21 -Dtests.fips.enabled=true
```

**Applicable branches:**
9.1

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dsearch.vectors%2F40_knn_search%2FDimensions%20are%20dynamically%20set%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [search.vectors/40_knn_search:632]: field [test_index.mappings.properties.embedding.dims] is null
```

**Issue Reasons:**
- [9.1] 2 failures in test test {p0=search.vectors/40_knn_search/Dimensions are dynamically set} (3.2% fail rate in 62 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -519,9 +519,6 @@ tests:
 - class: org.elasticsearch.xpack.monitoring.exporter.http.HttpExporterTests
   method: testCreateRestClient
   issue: https://github.com/elastic/elasticsearch/issues/130600
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=search.vectors/40_knn_search/Dimensions are dynamically set}
-  issue: https://github.com/elastic/elasticsearch/issues/130626
 - class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
   method: test {match-operator.MatchWithMoreComplexDisjunctionAndConjunction SYNC}
   issue: https://github.com/elastic/elasticsearch/issues/130640
@@ -537,9 +534,6 @@ tests:
 - class: org.elasticsearch.test.rest.yaml.RcsCcsCommonYamlTestSuiteIT
   method: test {p0=search.highlight/50_synthetic_source/text single fvh source order}
   issue: https://github.com/elastic/elasticsearch/issues/130657
-- class: org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT
-  method: test {yaml=search.vectors/40_knn_search/Dimensions are dynamically set}
-  issue: https://github.com/elastic/elasticsearch/issues/130661
 - class: org.elasticsearch.test.rest.yaml.RcsCcsCommonYamlTestSuiteIT
   method: test {p0=search.vectors/41_knn_search_half_byte_quantized/Test create, merge, and search cosine}
   issue: https://github.com/elastic/elasticsearch/issues/130663

